float32 rvecs
# float32 y
# float32 z
